cli/root.go

```diff
@@ -83,10 +110,78 @@ func init() {
 		// override flags if exist
 		cfg.Flags = override
 
-		// register config
-		err := Container.Register(cfg)
+		endureCfg := initEndureConfig()
+		if endureCfg == nil {
+			panic("endure config should not be nil")
+		}
+
+		var lvl endure.Level
+
+		switch endureCfg.LogLevel {
+		case "debug":
+			lvl = endure.DebugLevel
+		case "info":
+			lvl = endure.InfoLevel
+		case "warning":
+			lvl = endure.WarnLevel
+		case "error":
+			lvl = endure.ErrorLevel
+		case "panic":
+			lvl = endure.PanicLevel
+		case "fatal":
+			lvl = endure.FatalLevel
+		default:
+			panic(fmt.Sprintf("unknown log level, provided: %s, availabe: debug, info, warning, error, panic, fatal\n", endureCfg.LogLevel))
+		}
+
+		var err error
+		if endureCfg.PrintGraph {
+			Container, err = endure.NewContainer(nil, endure.SetLogLevel(lvl), endure.RetryOnFail(endureCfg.RetryOnFail), endure.SetStopTimeOut(endureCfg.GracePeriod), endure.Visualize(endure.StdOut, ""))
+		} else {
+			Container, err = endure.NewContainer(nil, endure.SetLogLevel(lvl), endure.RetryOnFail(endureCfg.RetryOnFail), endure.SetStopTimeOut(endureCfg.GracePeriod))
+		}
+
 		if err != nil {
-			panic(err)
+			log.Fatal(err)
+		}
```

Comment:
Can this section be encapsulated from main.go so it's easier to read for engineers who want to make their own build?

Reply:
Unfortunately no. To configure endure we use a path to the config, which is populated only in the `root.go`. For the engineers, there is zero difference where to put plugins. Just need to be covered in the documentaton.